tests/foundationals/dinov2/test_weights.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+import torch
+from refiners.foundationals.dinov2.dinov2 import (
+    DINOv2_base,
+    DINOv2_base_reg,
+    DINOv2_large,
+    DINOv2_large_reg,
+    DINOv2_small,
+    DINOv2_small_reg,
+)
+from refiners.foundationals.dinov2.vit import ViT
+
+from scripts.conversion.convert_dinov2 import convert_dinov2_facebook
+
+TORCH_HUB_REPO = "facebookresearch/dinov2"
+ATOL = 1e-4
+SEED = 0
+
+PAIRS = (
+    ("dinov2_vits14", DINOv2_small),
+    ("dinov2_vits14_reg", DINOv2_small_reg),
+    ("dinov2_vitb14", DINOv2_base),
+    ("dinov2_vitb14_reg", DINOv2_base_reg),
+    ("dinov2_vitl14", DINOv2_large),
+    ("dinov2_vitl14_reg", DINOv2_large_reg),
+    # TODO: implement SwiGLU layer
+    # ("dinov2_vitg14", DINOv2_giant),
+    # ("dinov2_vitg14_reg", DINOv2_giant_reg),
+)
+
+
+@pytest.mark.parametrize("fb_name, rf_classname", PAIRS)
+def test_dinov2_facebook_weights(fb_name: str, rf_classname: ViT) -> None:
+    torch.manual_seed(SEED)
+    input_data = torch.randn(2, 3, 518, 518)
+
+    # get output from facebook model
+    fb_model = torch.hub.load(TORCH_HUB_REPO, fb_name)
```

Comment:
It's not ideal to use `torch.hub.load` here but unfortunately there is no (even unofficial) pypi package (unlike, e.g. https://pypi.org/project/segment-anything-py/ that we use in `test_sam.py`). At least you could:

1. Set the cache dir via `torch.hub.set_dir(torch_hub_cache)` and define a new fixture for it (could go under `conftest.py`):

```python
@pytest.fixture(scope="session")
def torch_hub_cache(test_weights_path: Path) -> Path:
    return test_weights_path / "torch_hub"
```

2. Use the "local" mode to refer to a local directory (`torch.hub.load("/path/to/dinov2", "dinov2_vits14", source="local")`)
3. Use a fixture to refer to this local directory, e.g.:

```python
# in conftest.py
@fixture(scope="session")
def torch_hub_path() -> Path:
    from_env = os.getenv("REFINERS_TORCH_HUB_DIR")
    return Path(from_env) if from_env else PARENT_PATH / "torch_hub"
```

And:

```python
@pytest.fixture(scope="module")
def dinov2_repo_path(torch_hub_path: Path) -> Path:
    repo_path = torch_hub_path / "dinov2"
    if not repo_path.is_dir():
        warn(f"could not dinov2 repo at {repo_path}, skipping")
        pytest.skip(allow_module_level=True)
    return repo_path
```

And we'll have test setup instructions to pre-clone the repo